### PR TITLE
Vulkan: Speed up and simplify hardware tesselation by using shader storage buffers

### PR DESCRIPTION
**Pull request by hrydgard**
No more temp textures. This is quite a bit faster, especially on mobile.

### CODE REVIEW

File: GPU/Common/SplineCommon.cpp [hunk 1]
```diff
@@ -908,13 +908,22 @@ void DrawEngineCommon::SubmitSpline(const void *control_points, const void *indi
...
-			if (hasColor)
-				memcpy(col + idx * 4, Vec4f::FromRGBA(points[idx]->color_32).AsArray(), 4 * sizeof(float));
+			memcpy(p, points[idx]->pos.AsArray(), 3 * sizeof(float));
+			p += posStride;
+			if (hasTexCoords) {
+				memcpy(t, points[idx]->uv, 2 * sizeof(float));
+				t += texStride;
+			}
+			if (hasColor) {
+				memcpy(c, Vec4f::FromRGBA(points[idx]->color_32).AsArray(), 4 * sizeof(float));
```
Comment (unknownbrackets) [inline GPU/Common/SplineCommon.cpp]: Dumb question: in theory, could we use `VK_FORMAT_R8G8B8A8_UNORM` for this, in some way, to communicate less data?  Or would that probably not help anyway (I guess especially with that padding issue)?

-[Unknown]

Comment (hrydgard) [reply GPU/Common/SplineCommon.cpp]: Could stuff it in an uint32 and expand to float in the shader. Not worth the trouble, it's so little data anyway.

Comment (unknownbrackets) [reply GPU/Common/SplineCommon.cpp]: Hmm, I thought Final Fantasy 4 used very high u/v, such that we had to keep increasing a buffer size, even though simple implementations just use 4x4 usually.  Maybe that was only generated points...?

Looks like the upper bound in size is about 3MB (65025 points - might be in PSP RAM more compressed using indices and s16.)

-[Unknown]

Comment (hrydgard) [reply GPU/Common/SplineCommon.cpp]: IIRC that's it, it used crazy tesselation levels but not that much input data.

Haven't tried FF4 with this code, feel free to have a go :)

Comment (unknownbrackets) [reply GPU/Common/SplineCommon.cpp]: Heh, there are a few obvious graphical issues in FF4, good thing I tried (not spline related.)

-[Unknown]

Comment (hrydgard) [reply GPU/Common/SplineCommon.cpp]: Ah, indeed. It's a stencil issue, I got it.

**Pull request merged**